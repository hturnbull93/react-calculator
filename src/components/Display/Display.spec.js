import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Display from "./Display";

describe("Display", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Display displayValue={""} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a div", () => {
    const wrapper = shallow(<Display displayValue={""} />);
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("renders the value of displayValue", () => {
    const wrapper = shallow(<Display displayValue={""} />);
    wrapper.setProps({ displayValue: "test" });
    expect(wrapper.text()).toEqual("test");
  });
});
