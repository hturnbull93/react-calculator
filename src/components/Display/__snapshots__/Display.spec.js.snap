// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="display-container"
>
  <p
    className="display-value"
  >
    
  </p>
</div>
`;
